apps/workers/openaiWorker.ts

```diff
@@ -166,6 +180,47 @@ async function inferTagsFromImage(
   );
 }
 
+async function inferTagsFromPDF(
+  jobId: string,
+  bookmark: NonNullable<Awaited<ReturnType<typeof fetchBookmark>>>,
+  inferenceClient: InferenceClient,
+) {
+  const { asset } = await readAsset({
+    userId: bookmark.userId,
+    assetId: bookmark.asset.assetId,
+  });
+  if (!asset) {
+    throw new Error(
+      `[inference][${jobId}] AssetId ${bookmark.asset.assetId} for bookmark ${bookmark.id} not found`,
+    );
+  }
+  const pdfParse = await readPDFText(asset);
+  if (!pdfParse?.text) {
+    throw new Error(
+      `[inference][${jobId}] PDF text is empty. Please make sure that the PDF includes text and not just images.`,
+    );
+  }
+
+  await db
+    .update(bookmarkAssets)
+    .set({
+      content: pdfParse.text,
+      info: pdfParse.info ? JSON.stringify(pdfParse.info) : null,
+      metadata: pdfParse.metadata ? JSON.stringify(pdfParse.metadata) : null,
+    })
+    .where(eq(bookmarkAssets.id, bookmark.id));
```

Comment:
I think this logic fits more into being in the `crawler`. It's the worker type that enriches the `bookmark` in prep for the inference. We should probably rename it as well now that it's no longer about "crawling" :D 

I don't mind keeping this here for the sake of this PR and I can move it after it lands :)

Reply:
I would prefer to move it in the next PR to make this one stable.